### DISCUSSION TITLE
What are smart contracts for doge ?

### DISCUSSION THREAD

**Discussion by godcloutier** [godcloutier/officialdogevm#11] [General]
You may be aware of smart contracts already because of Ethereum or Binance smart chain. If not here is a simple definition:

Smart contracts are a way to connect trustless applications or dapp / decentralized applications to a blockchain and ensure its validity via the blockchain protocol or consensus.

Per example, NFTs are powered by smart contracts. 

### But why not use existing smart contracts like the EVM (ethereum, binance and others use it) for doge? 

Since their availability on the Ethereum blockchain (circa 2015) smart contracts did allow a lot of innovation but also created a lot of problems in the cryptocurrencies ecosystem. 

The fact that it was too easy to make a token did open the door to many scams that we will not name in the context of this Q&A. 

Also, DOGE is not and will never be a utility chain. DOGE is transactional and is "dog money". So the need of a one size fit all solution is not appropriate for the DOGE blockchain. Our community did work hard into supporting the Doge branding and we do not want any outside force to temper with it. Also one of our motto is : in doge we trust. We love Doge! Why would we want to put tokens on it ? 

For that reason, we need to develop a new kind of smart contracts targeted to Doge utility which means: optimize transactional operations, allow anonymization of data for per example medical patient files, allow NFTs to be created and custom utility via the possibility to code "plugins" which will be  an integrated service. The only caveat is that the said plugin will need to be approved first by the developers. 

HOWEVER, that approval also has many advantages for the plugin makers. It ensure them and the whole community that a proper audit will be made before those applications are put on the chain avoiding per example the bloating of the chain itself (some will remember the cryptokitty problem on ethereum ;) )

While auditing those applications it will also bring innovation to the chain a we will be able to see what needs really are.

Another aspect we want to work on is make it easier for merchants to integrate doge to their POS / Online store. 

If you have more questions feel free to send them to us or comment on this discussion.

**Comment by fmhc** [comment]
![image](https://user-images.githubusercontent.com/3387385/120711800-1630e480-c4c0-11eb-975f-a4124b7d0700.png)
this is an architectural draft of how we could get it together

**Reply by godcloutier** [reply]
I like it but eh really not sure about the Solana part tho too much work to be done I think to make it that way other than that it looks perfect good work now let's get to work and test that thing :)

**Reply by fmhc** [reply]
the dotted boxes are just extension parts. I think the main work to be done is having a PoC to write extended data according to a smartdoge standard and then have a machine encoding that and processing some smart contract magic. I'm happy with sending and receiving a NFT standard to build upon that; and then have a basic smart contract machine we can build upon.

smartdoge analytics --> on-chain analytics based on interpreted / transformed raw messages

monitoring and load balancing for the VMs will also be a thing, depending on the architectural integration we end up with... but lets see. The green parts are our baseline for a proof of concept, with smartdoge tx standard and a routing engine with a dummy endpoint VM would be next milestone.

i updated some parts regarding a first goal for PoC, have putten in some thoughts from recent twitter conversations.
![image](https://user-images.githubusercontent.com/3387385/120729968-b860c480-c4e0-11eb-81d1-182ee7cf5619.png)

we will probably update this in the upcoming days a thousand times, so as of writing this, the picture is outdated :D
